alfasim_sdk/types.py

```diff
@@ -113,11 +113,28 @@ def check(self, attr: Attribute, values: List[str]) -> None:  # pylint: disable=
 @attr.s(kw_only=True)
 class Reference(BaseField):
     ref_type = attrib()
+    container_type = attrib(default=None, validator=[optional(instance_of(str)), optional(check_string_is_not_empty)])
+
+    def __attrs_post_init__(self):
+        if issubclass(self.ref_type, ALFAsimType):
+            self.container_type = attr.fields(self.ref_type).container_type.default
```

Comment:
Here check that `self.container_type is not None` and raise an error otherwise. 👍

Reply:
Sorry I didn't catch what you mean it by `check that is not None` and raise if it's `None`

```python
if self.container_type is not None:
    raise TypeError("A container_type should not be given when an ALFAsimType is given)
self.container_type = self.ref_type._CONTAINER_TYPE
```
or
```python
self.container_type = self.ref_type._CONTAINER_TYPE
if self.container_type is None:
    raise Type("Something went wrong, did you forgot to specify _CONTAINER_TYPE?")